src/mesido/potential_errors.py

```diff
@@ -0,0 +1,67 @@
+from enum import Enum
+from typing import Dict, List, NoReturn
+
+from mesido.exceptions import MesidoAssetIssueError
+
+AssetId = str
+ErrorMessage = str
+
+
+# Asset error type that can occur-- move to error potential_err.....
+class MesidoAssetIssueType(Enum):
+    HEAT_DEMAND_POWER = "heat_demand.power"
+    COLD_DEMAND_POWER = "cold_demand.power"
+    HEAT_DEMAND_TYPE = "heat_demand.type"
+
+
+class PotentialErrors:
+    """Singleton, do not instantiate. Use POTENTIAL_ERRORS."""
+
+    _gathered_potential_issues: Dict[MesidoAssetIssueType, Dict[AssetId, List[ErrorMessage]]]
+
+    def __init__(self):
+        self._gathered_potential_issues = {}
+
+    def add_potential_issue(
+        self, issue_type: MesidoAssetIssueType, asset_id: AssetId, error_message: ErrorMessage
+    ) -> None:
+        """
+        Add potential issues to _gathered_potential_issues.
+
+        """
+        self._gathered_potential_issues.setdefault(issue_type, {})
+        self._gathered_potential_issues[issue_type][asset_id] = error_message
+
+    def have_issues_for(self, issue_type: MesidoAssetIssueType) -> bool:
+        """
+        Check if the potential issue exists.
+
+        """
+        result = False
```

Comment:
Alternatively, this could be a oneliner:
```python
return issue_type in self._gathered_potential_issues
```

Reply:
I prefer the current setup